src/mds/Server.cc

```diff
@@ -5971,7 +5997,16 @@ int Server::parse_quota_vxattr(string name, string value, quota_info_t *quota)
       }
     } else if (name == "quota.max_bytes") {
       string cast_err;
-      int64_t q = strict_iec_cast<int64_t>(value, &cast_err);
+      std::pair<int64_t, string> act_value = make_float_max_byte_parseable(value);
+      if (act_value.first == -CEPHFS_EINVAL) {
+        dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
+        << act_value.second << dendl;
+        return -CEPHFS_EINVAL;        
+      }
+      int64_t q = strict_iec_cast<int64_t>(act_value.second, &cast_err);
+      if (act_value.first != -1)  {
+        q = q / (std::pow(10, act_value.first));
+      }
```

Comment:
Oh here's where you're doing the magic of converting 15K to 1.5K (for e.g. `setfattr -n ceph.quota.max_bytes -v 1.5K ./mnt/foo/`) 🙂

Umm, why to do this way? I mean you make make it more modular by having this code in the func `Server::make_float_max_byte_parseable` itself. Then all you need to do is call the func and check if it returned some errno then 
```
 if(!cast_err.empty()) {
        dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
        << cast_err << dendl;
```
(Just like the existing code).

Reply:
Another reason is IMO it doesn't make sense to separate the code logic.